### DISCUSSION TITLE
Other results during creation with api

### DISCUSSION THREAD

**Discussion by JangDongHe** [Mikubill/sd-webui-controlnet#2315] [Q&A] [answered]
Hello, I am a student studying for developers in Korea.

I am working on combining qr code and image using the control net

But if I use api, I get a different result than what I generated in webui, so can you help me what's the problem?

I've tried different types of payloads by looking at various api use cases, but I'm experiencing a phenomenon that the base qr image is not applicable.

If you used api
![00011-3613720951](https://github.com/Mikubill/sd-webui-controlnet/assets/118067618/8d1d0bdc-32a0-4063-9105-6e2b13dcffe1)

Create from webui
![model1_img](https://github.com/Mikubill/sd-webui-controlnet/assets/118067618/2808e657-e769-48b0-8bcc-a2c124fa16ac)

The payload is as follows.

String payload = "{\r\n"
                + "    "init_images": ["data:image/png;base64,"+ base64QR + ""],\r\n"
                 + "    "prompt": "" + positivePrompt + "",\r\n"
                 + "    "negative_prompt": "" + negativePrompt + "",\r\n"
                 + "    "sd_model_checkpoint": "" + dto.getCheckPoint() + "",\r\n"
                 + "    "denoising_strength": " + dto.getDenoisingStrength() + ",\r\n"
                 + "    "cfg_scale": "+dto.getCfgScale()+",\r\n"
                 + "    "sampler_name": "DPM++ 2M Karras",\r\n"
                 + "    "width": 768 ,\r\n"
                 + "    "height": 768,\r\n"
                 + "    "steps": 20,\r\n"
                 + "    "alwayson_scripts": {\r\n"
                 + "        "controlnet": {\r\n"
                 + "            "args": [\r\n"
                 + "                {\r\n"
                 + "                    "enabled": "true",\r\n"
                 + "                    "module": "tile_resample",\r\n"
                 + "                    "model": "control_v11f1e_sd15_tile [a371b31b]",\r\n"
                 + "                    "weight": 0.9,\r\n"
                 + "                    "resize_mode": "Crop and Resize",\r\n"
                 + "                    "threshold_a": 1,\r\n"
                 + "                    "guidance_start": 0.23,\r\n"
                 + "                    "guidance_end": 1,\r\n"
                 + "                    "control_mode": "Balanced",\r\n"
                 + "                    "pixel_perfect": "true"\r\n"
                 + "                }\r\n"
                 + "            ]\r\n"
                 + "        }\r\n"
                 + "    }\r\n"
                 + "}";

**Comment by JangDongHe** [comment]
{
    "init_images": [
        "data:image/png;base64, base64QR "
    ],
    "prompt": " positivePrompt ",
    "negative_prompt": " negativePrompt ",
    "sd_model_checkpoint": " dto.getCheckPoint() ",
    "denoising_strength": " dto.getDenoisingStrength() ",
    "cfg_scale": "dto.getCfgScale()",
    "sampler_name": "DPM++ 2M Karras",
    "width": 768,
    "height": 768,
    "steps": 20,
    "alwayson_scripts": {
        "controlnet": {
            "args": [
                {
                    "enabled": "true",
                    "module": "tile_resample",
                    "model": "control_v11f1e_sd15_tile [a371b31b]",
                    "weight": 0.9,
                    "resize_mode": "Crop and Resize",
                    "threshold_a": 1,
                    "guidance_start": 0.23,
                    "guidance_end": 1,
                    "control_mode": "Balanced",
                    "pixel_perfect": "true"
                }
            ]
        }
    }
}

**Comment by JangDongHe** [comment]
This setting is used by webui
![set1](https://github.com/Mikubill/sd-webui-controlnet/assets/118067618/973ff19b-8082-47c8-9552-c8e967c726b6)
![set2](https://github.com/Mikubill/sd-webui-controlnet/assets/118067618/bd5eed54-e292-495f-a1cc-062bb343a4c4)

**Comment by huchenlei** [comment, marked as answer]
You can use https://github.com/huchenlei/sd-webui-api-payload-display to dump out API payload from your A1111 generations.

**Reply by JangDongHe** [reply]
Thank you for your answer. This answer is very useful

**Reply by ammartaher** [reply]
How can I use this payload display @huchenlei ? there is no manual or so.

**Reply by huchenlei** [reply]
See https://github.com/huchenlei/sd-webui-api-payload-display#usage

**Reply by ammartaher** [reply]
**Ignore this comment: I found out i was stupid enough not to understand it is an extension for SD and A1111 will take care of showing it:D).** 
it says it will show up in the accordion, does that mean i need to add it to the stable diffusion folder soemwhere to be loaded? like an extension or so?

**Reply by ammartaher** [reply]
<img width="556" alt="image" src="https://github.com/Mikubill/sd-webui-controlnet/assets/39812337/14b11683-3424-420a-ba70-80c1f674220f">
I installed the extension but it doesn't show anything. it is supposed to show whatever i do from an external app calling the api right?

**Reply by ammartaher** [reply]
Figured out, the vest way is to adjust your controllers to a good result on Webui and then copy tye payload from that extension to your api sender app and adjust it accordingly. Works like magic. Thanks anyway :))
